types/animation.d.ts

```diff
@@ -0,0 +1,33 @@
+import { PlayerObject } from './model'
+
+export type Animation = CompositeAnimation | typeof WalkAnimation
+
+declare function invokeAnimation(
+  animation: Animation,
+  player: PlayerObject,
+  time: number
+): void
+
+declare class AnimationHandle {
+  animation: Animation
+  paused: boolean
+  speed: number
+
+  constructor(animation: Animation)
+
+  play(player: PlayerObject, time: number): void
+
+  reset(): void
+}
+
+export class CompositeAnimation {
+  handles: Set<AnimationHandle>
```

Comment:
Readonly here.

Reply:
Will it be `private` better?

Reply:
You're right. I think `add()` and `remove()` are already enough.